uk_bin_collection/uk_bin_collection/councils/LancasterCityCouncil.py

```
@@ -57,30 +70,48 @@ def parse_data(self, page: str, **kwargs) -> dict:
-                                    "type": bin_type_element.text.strip().removesuffix(" Collection Service"),
+                                    "type": collection_type,
                                     "collectionDate": dt.strftime(date_format),
                                 }
                             )
-                    except (ValueError, AttributeError) as e:
+                    except ValueError:
                         # Skip invalid date or missing elements
                         continue
 
+        data["bins"].sort(
+            key=lambda x: datetime.strptime(x.get("collectionDate"), date_format)
```

Ruff (0.15.6)

[error] `date_format` may be undefined, or defined from star imports

(F405)